### PR TITLE
Calculate share button padding dynamically for scroll affordance

### LINKED ISSUE #11898: Share buttons do not wrap if beyond viewport width
In amp-story, when there are more share buttons than viewport width, the buttons do not wrap, and either are cut off or do not display on the screen.

Example: 6th button beyond screen width

amp-story-share-buttons

Example: Narrow screen, button cut off
amp-story-share-buttons-narrow-screen

### PR DESCRIPTION
**Pull request by alanorozco**
Fixes #11898.

When the share buttons won't fit in the container, thus making it scrollable, we need a visual affordance so that the user realizes that the element is scrollable.

This change allows the widget to calculate the padding so that there's always an element cut-off if needed, otherwise allows the buttons to have "fluid" padding so that they fit the container.

scrollafordance

(This could be done with CSS/flexbox **IFF** the element wasn't scrollable.)

### CODE REVIEW

File: extensions/amp-story/0.1/share.js [hunk 1]
```diff
@@ -195,6 +229,86 @@ export class ShareWidget {
...
+          state.noop = true;
+          return;
+        }
+
+        const leftMargin =
+            dev().assert(items[0].firstElementChild)./*OK*/offsetLeft -
+            this.root_./*OK*/offsetLeft;
+
+        const iconWidth =
+            dev().assert(items[0].firstElementChild)./*OK*/offsetWidth;
```
Comment (choumx) [inline extensions/amp-story/0.1/share.js]: Nit: Extract `items[0].firstElementChild` into a local var.

File: extensions/amp-story/0.1/share.js [hunk 2]
```diff
@@ -195,6 +229,86 @@ export class ShareWidget {
...
+
+        // Total width that the buttons will occupy with minimum padding
```
Comment (choumx) [inline extensions/amp-story/0.1/share.js]: Super-nit: Trailing period.

File: extensions/amp-story/0.1/share.js (hunk 1 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/share.js]: Done.

File: extensions/amp-story/0.1/share.js (hunk 2 above)
Comment (alanorozco) [reply extensions/amp-story/0.1/share.js]: Done.

**Pull request merged**